File: backend/src/models/Review.js
/**
 * Review Model - Customer reviews for foods and restaurants
 */
const mongoose = require('mongoose');

const reviewSchema = new mongoose.Schema({
  user: { type: mongoose.Schema.Types.ObjectId, ref: 'User', required: true },
  food: { type: mongoose.Schema.Types.ObjectId, ref: 'Food' },
  restaurant: { type: mongoose.Schema.Types.ObjectId, ref: 'Restaurant' },
  hotel: { type: mongoose.Schema.Types.ObjectId, ref: 'User' },
  order: { type: mongoose.Schema.Types.ObjectId, ref: 'Order' },
  rating: { type: Number, required: true, min: 1, max: 5 },
  comment: { type: String, default: '' },
  images: [{ type: String }],
  // Engagement
  likeCount: { type: Number, default: 0 },
  likedBy: [{ type: mongoose.Schema.Types.ObjectId, ref: 'User' }],
  isVerifiedPurchase: { type: Boolean, default: false }
}, { timestamps: true });

// Index for efficient queries
reviewSchema.index({ food: 1, createdAt: -1 });
reviewSchema.index({ hotel: 1, createdAt: -1 });
reviewSchema.index({ user: 1 });

module.exports = mongoose.model('Review', reviewSchema);

File: frontend/lib/data/models/review.dart
/// Review Model
class Review {
  final String id;
  final String oderId;
  final String foodId;
  final String userId;
  final String userName;
  final int rating;
  final String comment;
  final List<String> images;
  final DateTime createdAt;

  Review({
    required this.id,
    required this.oderId,
    required this.foodId,
    required this.userId,
    required this.userName,
    required this.rating,
    required this.comment,
    this.images = const [],
    required this.createdAt,
  });

  factory Review.fromJson(Map<String, dynamic> json) {
    return Review(
      id: json['_id'] ?? '',
      oderId: json['order']?.toString() ?? '',
      foodId: json['food']?.toString() ?? '',
      userId:
          json['user']?['_id']?.toString() ?? json['user']?.toString() ?? '',
      userName: json['user']?['name'] ?? 'Anonymous',
      rating: json['rating'] ?? 0,
      comment: json['comment'] ?? '',
      images: List<String>.from(json['images'] ?? []),
      createdAt: DateTime.tryParse(json['createdAt'] ?? '') ?? DateTime.now(),
    );
  }
}
